src/simplewallet/simplewallet.cpp

```diff
@@ -2202,24 +2202,36 @@ bool simple_wallet::transfer_main(bool new_algorithm, const std::vector<std::str
     if (m_wallet->always_confirm_transfers() || ptx_vector.size() > 1)
     {
         uint64_t total_fee = 0;
+        uint64_t dust_not_in_fee = 0;
+        uint64_t dust_in_fee = 0;
         for (size_t n = 0; n < ptx_vector.size(); ++n)
         {
           total_fee += ptx_vector[n].fee;
+
+          if (ptx_vector[n].dust_added_to_fee)
+            dust_in_fee += ptx_vector[n].dust;
+          else
+            dust_not_in_fee += ptx_vector[n].dust;
         }
 
-        std::string prompt_str;
+        std::stringstream prompt;
         if (ptx_vector.size() > 1)
         {
-          prompt_str = (boost::format(tr("Your transaction needs to be split into %llu transactions.  "
-            "This will result in a transaction fee being applied to each transaction, for a total fee of %s.  Is this okay?  (Y/Yes/N/No)")) %
-            ((unsigned long long)ptx_vector.size()) % print_money(total_fee)).str();
+          prompt << boost::format(tr("Your transaction needs to be split into %llu transactions.  "
+            "This will result in a transaction fee being applied to each transaction, for a total fee of %s")) %
+            ((unsigned long long)ptx_vector.size()) % print_money(total_fee);
         }
         else
         {
-          prompt_str = (boost::format(tr("The transaction fee is %s.  Is this okay?  (Y/Yes/N/No)")) %
-            print_money(total_fee)).str();
+          prompt << boost::format(tr("The transaction fee is %s")) %
+            print_money(total_fee);
         }
-        std::string accepted = command_line::input_line(prompt_str);
+        if (dust_in_fee != 0) prompt << boost::format(tr(", of which %s is dust from change")) % print_money(dust_in_fee);
+        if (dust_not_in_fee != 0)  prompt << tr(".") << ENDL << boost::format(tr("A total of %s from dust change will be sent to dust address")) 
+                                                   % print_money(dust_not_in_fee);
```

Comment:
I guess it doesn't need to be handled separately at the moment.  

It's just that addr_for_dust exists as a field at the moment in tx_dust_policy, which allows for the possibility of a dust address in the future.  It seems that this field is ignored at the moment though.

My preference would probably be to remove dust and dust policy and just have everything go back as change, which would fix the bug and (as per your comment below) make things less confusing.  However I'm a bit new to the code so I thought there might be a good reason why the dust handling was there in the first place.